lib/active_merchant/billing/gateways/jetpay_v2.rb

```diff
@@ -0,0 +1,406 @@
+        "903" =>  "Debit not supported.",
+        "904" =>  "Private label not supported.",
+        "905" =>  "Invalid card type.",
+        "906" =>  "Unit not active.",
+        "908" =>  "Manual card entry invalid.",
+        "909" =>  "Invalid track information.",
+        "911" =>  "Master merchant not found.",
+        "912" =>  "Invalid card format.",
+        "913" =>  "Invalid card type.",
+        "914" =>  "Invalid card length.",
+        "917" =>  "Expired card.",
+        "919" =>  "Invalid entry type.",
+        "920" =>  "Invalid amount.",
+        "921" =>  "Invalid messge format.",
+        "923" =>  "Invalid ABA.",
+        "924" =>  "Invalid DDA.",
+        "925" =>  "Invalid TID.",
+        "926" =>  "Invalid Password.",
+        "930" =>  "Invalid zipcode.",
+        "931" =>  "Invalid Address.",
+        "932" =>  "Invalid ZIP and Address.",
+        "933" =>  "Invalid CVV2.",
+        "934" =>  "Program Not Allowed.",
+        "935" =>  "Invalid Device/App.",
+        "940" =>  "Record Not Found.",
+        "941" =>  "Merchant ID error.",
+        "942" =>  "Refund Not Allowed.",
+        "943" =>  "Refund denied.",
+        "955" =>  "Invalid PIN block.",
+        "956" =>  "Invalid KSN.",
+        "958" =>  "Bad Status.",
+        "959" =>  "Seek Record limit exceeded.",
+        "960" =>  "Internal Key Database Error.",
+        "961" =>  "TRANS not Supported. Cash Disbursement required a specific MCC.",
+        "962" =>  "Invalid PIN key (Unknown KSN).",
+        "981" =>  "Invalid AVS.",
+        "987" =>  "Issuer Unavailable.",
+        "988" =>  "System error SD.",
+        "989" =>  "Database Error.",
+        "992" =>  "Transaction Timeout.",
+        "996" =>  "Bad Terminal ID.",
+        "997" =>  "Message rejected by association.",
+        "999" =>  "Communication failure",
+        nil   =>  "No response returned (missing credentials?)."
+      }
+
+      def initialize(options = {})
+        requires!(options, :login)
+        super
+      end
+
+      def purchase(money, credit_card, options = {})
+        commit(money, build_sale_request(money, credit_card, options))
+      end
+
+      def authorize(money, credit_card, options = {})
+        commit(money, build_authonly_request(money, credit_card, options))
+      end
+
+      def capture(money, reference, options = {})
+        split_authorization = reference.split(";")
+        transaction_id = split_authorization[0]
+        token = split_authorization[3]
+        commit(money, build_capture_request(transaction_id, money, options), token)
+      end
+
+      def void(reference, options = {})
+        transaction_id, approval, amount, token = reference.split(";")
+        commit(amount.to_i, build_void_request(amount.to_i, transaction_id, approval, token, options), token)
+      end
+
+      def credit(money, credit_card, options = {})
+        commit(money, build_credit_request('CREDIT', money, nil, credit_card, nil, options))
```

Comment:
Slight inconsistency, unless there's some logic nuance: 'CREDIT' is sent from here to the `build_redit_request` for both `refund` and general `credit` actions, but the transaction type for other actions is sent from their `build_` methods.